### ISSUE TITLE
How to reproduce results on the summarization task?

### ISSUE THREAD

**Issue by Silenthinker** [facebookresearch/fairseq-py#9]
Hi,
I am trying to reproduce experimental results on two text summarization tasks: DUC-2004 and Gigaword. Is it possible to provide an example of training a model on summarization? In addition, how can I set parameters so that ROUGE scores in the [paper](https://arxiv.org/abs/1705.03122) can be reproduced? Thanks.

**Comment by michaelauli** [comment]
We used the data from here: https://github.com/facebookarchive/NAMAS
The remaining details are in the paper, e.g., vocab of 30K, hidden size 256, batchsize 128 etc.

**Comment by WangLilian** [comment]
Hi,
I read the paper that additionally you require outputs to be at least 14 words long, but I think length of  much target summarizations may less than 14. And I use the training parameters as following:
--lr 0.25 --clip-norm 0.1 --dropout 0.1 --max-tokens 4000 --arch fconv_giga
The architecture of fconv_giga is the same as the paper said. 
May I ask for some suggestion?
Thank you very much!

**Comment by michaelauli** [comment]
This looks good. You may need to tune the learning rate and --max-tokens but other than that this looks right. Since publishing the paper we changed the batching strategy, so you may have to experiment in getting a good setting for this problem.
